### PR TITLE
ci: clean up workflow files

### PR DESCRIPTION
**Pull request by innocenzi**
This pull request cleans up workflow files:
- I made the task names consistent in casing
- I removed two seemingly unused workflows
- I fixed `coding-conventions.yml`, which was corrupted in a rebase
- I removed the updating of `CHANGELOG.md` in the GitHub release workflow, as that doesn't work with branch protections
- I took the opportunity to update `package.json` scripts

### CODE REVIEW

File: .github/workflows/coding-conventions.yml [hunk 1]
```diff
@@ -35,8 +35,8 @@ jobs:
       - name: Run eslint
         run: bun run lint
 
-  phpstan:
-    name: "Run Static Analysis: PHPStan"
+ phpstan:
+    name: "Run static analysis: phpstan"
```
Comment (aidan-casey) [inline .github/workflows/coding-conventions.yml:39]: It looks to me like you are adopting the brand casing for GitHub, Bun, etc. in this PR. We should probably do the same for __PHPStan__.

File: .github/workflows/coding-conventions.yml [hunk 2]
```diff
@@ -35,8 +35,8 @@ jobs:
       - name: Run eslint
```
Comment (aidan-casey) [inline .github/workflows/coding-conventions.yml:35]: It looks to me like you are adopting the brand casing for GitHub, Bun, etc. in this PR. We should probably do the same for __ESLint__.

File: .github/workflows/coding-conventions.yml (hunk 1 above)
Comment (innocenzi) [reply .github/workflows/coding-conventions.yml]: Absolutely. Good catch!

**Comment by aidan-casey** [comment]
@innocenzi looks like there are still a few cases of the brand names being off (another phpstan, etc.). Once that's cleaned up, I like it and let's get this merged.

Side note, this is one of those things we should probably document in contributing.

**Comment by innocenzi** [comment]
@aidan-casey Thanks! Here I was, thinking I was clever with my perfectionism, and I missed one 😔

**Pull request merged**